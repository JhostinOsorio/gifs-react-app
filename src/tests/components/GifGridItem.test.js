const { shallow } = require("enzyme")
const { default: GifGridItem } = require("../../components/GifGridItem")

describe('Pruebas en GifGridItem', () => {

  const title = 'Es un titulo';
  const url = 'http://localhost:3000/static/logo192.png';
  const wrapper = shallow(<GifGridItem title={title} url={url}/>)

  test('debe mostrar <GifGridItem /> correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('debe tener un parrafo en el title', () => {
    const p = wrapper.find('p').text()
    expect(p.trim()).toBe(title)
  })
  
  test('debe de tener la imagen igual al url y alt de los props', () => {
    const img = wrapper.find('img')
    // console.log(img.props().src)
    expect(img.props().src).toBe(url)
    expect(img.props().alt).toBe(title)
  })

  test('debe tener animate__fadeIn', () => {
    const div = wrapper.find('div')
    expect(div.hasClass('animate__fadeIn')).toBe(true)
  })
  
})