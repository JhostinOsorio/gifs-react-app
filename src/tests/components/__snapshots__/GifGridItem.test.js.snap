// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Es un titulo"
    src="http://localhost:3000/static/logo192.png"
  />
  <p>
    Es un titulo
  </p>
</div>
`;
